.github/workflows/deploy_functions.yml

```diff
@@ -42,27 +49,54 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
+          ref: ${{ github.ref_name }}
           fetch-depth: 5
+
       - name: Cache node_modules
         uses: actions/cache@v2
         with:
           path: '**/node_modules'
           key: modules-${{ hashFiles('./yarn.lock') }}
 
+      - name: Generate and publish release
+        if: ${{ inputs.environment }} == 'development'
```

Comment:
Is that correct? shouldn't you test if the "base" branch is the main instead? or the master depending on the repository